sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormat.scala

```diff
@@ -93,6 +98,68 @@ class BinaryFileFormat extends FileFormat with DataSourceRegister {
 
     val pathGlobPattern = binaryFileSourceOptions.pathGlobFilter
 
+    val modificationTimeFilters = new mutable.ListBuffer[Long => Boolean]
```

Comment:
Discussed offline. Instead of building filter functions separately, both can be presented by `FileStatus => Boolean`